### PR TITLE
[mac] new OT API to control the beacon joinable flag

### PR DESCRIPTION
**Pull request by abtink**
This commit defines new methods in `Mac` along with new OpenThread
public APIs to allow direct control of the "Joining Permitted" flag
used in transmitted IEEE 802.15.4 Beacon frames.

A new spinel property `SPINEL_PROP_MAC_BEACON_JOINABLE` to control
this flags is added. Its get/set handlers are also implemented in NCP.
Spinel documentation is also updated for the new property.

### CODE REVIEW

File: src/core/mac/mac.hpp [hunk 1]
```diff
@@ -304,6 +304,23 @@ class Mac: public ThreadNetifLocator
     void SetBeaconEnabled(bool aEnabled) { mBeaconsEnabled = aEnabled; }
 
     /**
+     * This method gets the Joining Permitted flag used in any subsequently transmitted IEEE 802.15.4 Beacon frames.
+     *
+     * @retval TRUE   if the Joining Permitted flag will be set in Beacon frames.
+     * @retval FALSE  if the Joining Permitted flag will not set in Beacon frames.
+     *
+     */
+    bool GetBeaconJoinableFlag(void)  { return mBeaconJoinableFlag; }
```
Comment (jwhui) [inline src/core/mac/mac.hpp]: nits:
- change `Joinable` to `JoiningPermitted`?  Thread Spec calls this the "Joining Permitted" bit.
- declare as `const`

File: src/core/mac/mac.cpp [hunk 2]
```diff
@@ -750,10 +750,7 @@ void Mac::SendBeacon(Frame &aFrame)
     {
         beaconPayload->Init();
 
-        // set the Joining Permitted flag
-        GetNetif().GetIp6Filter().GetUnsecurePorts(numUnsecurePorts);
-
-        if (numUnsecurePorts)
+        if (mBeaconJoinableFlag)
```
Comment (jwhui) [inline src/core/mac/mac.cpp]: Joiner Router also needs to be updated to set/clear the Joining Permitted flag.

- https://github.com/openthread/openthread/blob/master/src/core/meshcop/joiner_router.cpp#L87
- https://github.com/openthread/openthread/blob/master/src/core/meshcop/joiner_router.cpp#L97

File: src/core/mac/mac.hpp (hunk 1 above)
Comment (abtink) [inline src/core/mac/mac.hpp]: Will do...

File: src/core/mac/mac.cpp (hunk 2 above)
Comment (abtink) [inline src/core/mac/mac.cpp]: Will add... also thanks for giving the link to the code... :)

**Comment by abtink** [comment]
Closing this for to combine it as a feature  with oob steering data...

**Pull request closed without merging**